elastic/beats
# [Metricbeat] Add vpc metricset for aws module

## What does this PR do?

This PR is to add support for collecting vpc related monitoring metrics from CloudWatch. This includes AWS/NATGateway, AWS/VPN and AWS/TransitGateway namespaces.

## Why is it important?

Users can use this vpc metricset to collect metrics related to their VPCs, for example: bytes received/sent through VPN tunnel, total number of concurrent active TCP connections through NAT gateway, and number of bytes sent/received from transit gateway.

## Checklist


## Author's Checklist

- [ ] generate data.json

## How to test this PR locally

Make sure the AWS account you use for testing has VPC related metrics first. 
Enable vpc metricset and then change aws.yml to:
```
- module: aws
  period: 1m
  credential_profile_name: elastic-test
  metricsets:
    - vpc
```

## Related issues
elastic/beats#14854
